docs/sphinx/user_guide/feature/reduction.rst

```diff
@@ -159,3 +159,153 @@ Reduction Policies
+
+.. note:: In the above example ``Res`` is a resource object that must be 
+          compatible with the ``EXEC_POL``. ``Seg`` is the iteration space
+          object for ``RAJA::forall``.
+
+.. important:: The order and types of the local reduction variables in the
+               kernel body lambda expression must match exactly with the
+               corresponding ``RAJA::expt::Reduce`` arguments to the
+               ``RAJA::forall`` to ensure that the correct result is obtained.
+
+RAJA::expt::ValLoc
+..................
+
+As with the current RAJA reduction interface, the new interface supports *loc* 
+reductions, which provide the ability to get a kernel/loop index at which the 
+final reduction value was found. With this new interface, *loc* reductions 
+are performed using ``ValLoc<T>`` types. Since they are strongly typed, they
+provide ``min()`` and ``max()`` operations that are equivalent to using 
+``RAJA_MIN()`` or ``RAJA_MAX`` macros as demonstrated in the code example below.
+Users must use ``getVal()`` and ``getLoc()`` methods to access the reduction 
+results::
+
+  double* a = ...;
+
+  using VL_DOUBLE = RAJA::expt::ValLoc<double>;
+  VL_DOUBLE rm_loc;
+      
+  RAJA::forall<EXEC_POL> ( Res, Seg, 
+  RAJA::expt::Reduce<RAJA::operators::minimum>(&rm_loc),
+  [=] (int i, VL_DOUBLE& _rm_loc) {
+    _rm_loc = RAJA_MIN(VL_DOUBLE(a[i], i), _rm_loc);  
+    //_rm_loc.min(VL_DOUBLE(a[i], i)); // Alternative to RAJA_MIN
+  }
+  );
+
+  std::cout << rm_loc.getVal() ...
+  std::cout << rm_loc.getLoc() ...
+
+Lambda Arguments
+................
+
+This interface takes advantage of C++ parameter packs to allow users to pass
+any number of ``RAJA::expt::Reduce`` objects to the ``RAJA::forall`` method::
+
+  double* a = ...;
+
+  using VL_DOUBLE = RAJA::expt::ValLoc<double>;
+  VL_DOUBLE rm_loc;
+  double rs;
+  double rm;
+        
+  RAJA::forall<EXEC_POL> ( Res, Seg, 
+    RAJA::expt::Reduce<RAJA::operators::plus>(&rs),        // --> 1 double added
+    RAJA::expt::Reduce<RAJA::operators::minimum>(&rm),     // --> 1 double added
+    RAJA::expt::Reduce<RAJA::operators::minimum>(&rm_loc), // --> 1 VL_DOUBLE added
+    RAJA::expt::KernelName("MyFirstRAJAKernel"),           // --> NO args added
+    [=] (int i, double& _rs, double& _rm, VL_DOUBLE& _rm_loc) {
+      _rs += a[i];
+      _rm = RAJA_MIN(a[i], _rm); 
+      _rm_loc.min(VL_DOUBLE(a[i], i));
+    }
+  );
+
+  std::cout << rs ...
+  std::cout << rm ...
+  std::cout << rm_loc.getVal() ...
+  std::cout << rm_loc.getLoc() ...
+
+Again, the lambda expression arguments are passed in the same order as
+the ``RAJA::expt::Reduce`` arguments to ``RAJA::forall``. Both the types and 
+order of arguments must match to get correct results and to compile 
+successfully. Otherwise, a static assertion will be triggered::
+
+  LAMBDA Not invocable w/ EXPECTED_ARGS.
+
+.. note:: This static assert is only enabled when passing a generic lambda.
```

Comment:
I'm not sure generic lambda is the phrasing you want to use here. Generic is used with lambda to mean using auto at the param type.

Reply:
I used generic as in "not extended". I believe the CUDA documentation refers to lambdas like this as well, so that was the convention I followed.

Reply:
How about non-device?

Reply:
non device lambdas are not considered extended lambdas by CUDA

Reply:
Wait, does generic lambda mean a host device lambda in cuda?

Reply:
I don't think it does. I think 'generic lambda' refers to the 'auto' param type(s) as you described. I believe that the CUDA 'extended lambda' concept refers to being able to decorate a lambda with __host__ __device__ and use them in either host or device execution context.